specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,567 @@
+                                        <text>Downloading this week's new music...</text>
+                                        <progress
+                                            title = "{progressTitle}
+                                            value = "{progressValue}"
+                                            valueStringOverride = "{progressValueString}"
+                                            status = "{progressStatus}" />
+                                    </binding>
+                                </visual>
+                            </toast>)";
+
+    XmlDocument doc;
+    doc.LoadXml(payload);
+    AppNotification notification(doc);
+    notification.Tag(tag);
+    notification.Group(group);
+
+    // Assign initial values for first notification progress UI
+    AppNotificationProgressData data;
+    data.Title(L"Weekly playlist"); // Binds to {progressTitle} in xml payload
+    data.Value(0.6); // Binds to {progressValue} in xml payload
+    data.ValueStringOverride(L"15/26 songs"); // Binds to {progressValueString} in xml payload
+    data.Status(L"Downloading..."); // Binds to {progressStatus} in xml payload
+    data.SequenceNumber(1);
+
+    notification.Progress(data);
+    AppNotificationManager::Default().Show(notification);
+}
+
+// Send subsequent progress updates
+winrt::Windows::Foundation::IAsyncAction UpdateProgressAsync()
+{
+    winrt::hstring tag = L"weekly-playlist";
+    winrt::hstring group = L"downloads";
+
+    // Assign new values
+    // Note that you only need to assign values that changed. In this example we don't assign progressStatus since we don't need to change it
+    AppNotificationProgressData data;
+    data.Value(0.7); // Binds to {progressValue} in xml payload
+    data.ValueStringOverride(L"18/26 songs"); // Binds to {progressValueString} in xml payload
+    data.SequenceNumber(2);
+
+    auto result = co_await AppNotificationManager::Default().UpdateProgressDataAsync(data, tag, group);
+    if (result != AppNotificationProgressResult::Succeeded)
+    {
+        LOG_HR_MSG(E_UNEXPECTED, "Notification Progress Update Failed!");
+    }
+}
+```
+# Remarks
+
+## Registration
+
+The developer should always call the Notification Registration API first to register the current process as
+the Activator target.
+
+## Foreground API calls
+
+The developer should always subscribe to AppNotification activation foreground events to intercept notification related actions if
+the app happens to be running in the foreground.
+
+## Manifest Registration
+
+For MSIX, the COM activator GUID and the exe need to be registered in the manifest. The launch args
+would need to be pre-set to a well-known string that defines Activation Triggers.
+
+```xml
+<Extensions>
+  <com:Extension Category="windows.comServer">
+    <com:ComServer>
+      <com:ExeServer Executable="SampleBackgroundApp\SampleBackgroundApp.exe" DisplayName="SampleBackgroundApp" Arguments="----AppNotificationActivationServer">
```

Comment:
when COM activates a LocalServer it passes [-Embedding command line](https://docs.microsoft.com/en-us/windows/win32/com/localserver32#remarks). is this argument used as well?

the sample does not show using either. is that missing?

Reply:
The reunion implementation will look for the keyword and parse the arguments accordingly